pkg/kbagent/service/command.go

```diff
@@ -143,34 +142,9 @@ func runCommandX(ctx context.Context, action *proto.ExecAction, parameters map[s
 		cmd.Env = mergedEnv
 	}
 
-	var (
-		stdin          io.WriteCloser
-		stdout, stderr io.ReadCloser
-	)
-	if stdinReader != nil {
-		var stdinErr error
-		stdin, stdinErr = cmd.StdinPipe()
-		if stdinErr != nil {
-			cancelTimeout()
-			return nil, errors.Wrapf(proto.ErrInternalError, "failed to create stdin pipe: %v", stdinErr)
-		}
-	}
-	if stdoutWriter != nil {
-		var stdoutErr error
-		stdout, stdoutErr = cmd.StdoutPipe()
-		if stdoutErr != nil {
-			cancelTimeout()
-			return nil, errors.Wrapf(proto.ErrInternalError, "failed to create stdout pipe: %v", stdoutErr)
-		}
-	}
-	if stderrWriter != nil {
-		var stderrErr error
-		stderr, stderrErr = cmd.StderrPipe()
-		if stderrErr != nil {
-			cancelTimeout()
-			return nil, errors.Wrapf(proto.ErrInternalError, "failed to create stderr pipe: %v", stderrErr)
-		}
-	}
+	cmd.Stdin = stdinReader
+	cmd.Stdout = stdoutWriter
+	cmd.Stderr = stderrWriter
 
 	errChan := make(chan error)
```

Comment:
I suggest using a buffered channel here (e.g. `make(chan error, 1)`). If nobody reads from the channel for some reason, the goroutine will be leaked.

Reply:
Solved. Due to branch naming issues, this PR will be closed. The new PR has the same content as this one.
https://github.com/apecloud/kubeblocks/pull/8168